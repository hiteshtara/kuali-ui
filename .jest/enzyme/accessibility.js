/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import assert from 'assert'
import { findDOMNode } from 'react-dom'
import { mount } from 'enzyme'
import axeCore from 'axe-core'

export async function assertComponentIsAccessible (component, axeConfig = {}) {
  const container = document.createElement('div')
  document.body.appendChild(container)

  const wrapper = mount(component, { attachTo: container })
  const nodeUnderTest = findDOMNode(wrapper.instance())

  const { violations = [] } = await axeCore.run(nodeUnderTest, axeConfig)

  for (const violation of violations) {
    const { help } = violation

    for (const node of violation.nodes) {
      const { target } = node
      assert.ok(false, `${help} on element ${target}`)
    }
  }

  document.body.removeChild(container)
}
