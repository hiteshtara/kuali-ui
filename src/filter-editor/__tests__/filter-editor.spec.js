/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import React from 'react'
import { shallow } from 'enzyme'

import FilterEditor from '..'

const noop = () => {}

describe('StandardLayout Component', () => {
  it('should render', () => {
    const wrapper = shallow(
      <FilterEditor
        columns={[]}
        fields={[]}
        filter={undefined}
        onClose={noop}
        onFilterChanged={noop}
        onFilterSaved={noop}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
