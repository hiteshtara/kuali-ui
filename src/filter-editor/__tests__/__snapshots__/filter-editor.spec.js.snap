// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StandardLayout Component should render 1`] = `
<Paper
  className="filter-editor"
  component="div"
  zDepth={1}
>
  <Content
    className=""
    tag="div"
  >
    <div
      className="filter-editor__header"
    >
      <Heading
        className="filter-editor__title"
        tag="h3"
      >
        Filter Rules
      </Heading>
    </div>
    <Rules
      className=""
    />
    <div
      className="filter-editor__actions"
    >
      <RaisedButton
        label="Add Rule"
        onClick={[Function]}
      >
        <Icon
          aria-hidden={true}
          iconClassName="material-icons"
          name="add_circle"
          variant="success"
        />
      </RaisedButton>
    </div>
  </Content>
  <IconButton
    className="filter-editor__close"
    onClick={[Function]}
    variant="plain"
  >
    <Icon
      aria-hidden={true}
      iconClassName="material-icons"
      name="clear"
      variant="default"
    />
  </IconButton>
</Paper>
`;
